ext/storage_sources/s3_store/s3_connection.cpp

```diff
@@ -11,51 +11,79 @@
 
 /*
  * ListBuckets --
- *     Builds a list of buckets from AWS account into a vector. Returns true if success, otherwise
- *     false.
+ *     Builds a list of buckets from AWS account into a vector. Returns 0 if success, otherwise
+ *     1.
  */
-bool
+int
 S3Connection::ListBuckets(std::vector<std::string> &buckets) const
 {
     auto outcome = m_S3CrtClient.ListBuckets();
     if (outcome.IsSuccess()) {
         for (const auto &bucket : outcome.GetResult().GetBuckets())
             buckets.push_back(bucket.GetName());
-        return true;
+        return (0);
     } else {
         std::cerr << "Error in ListBuckets: " << outcome.GetError().GetMessage() << std::endl
                   << std::endl;
-        return false;
+        return (1);
     }
 }
 
 /*
  * ListObjects --
- *     Builds a list of object names from a S3 bucket into a vector. Returns true if success,
- *     otherwise false.
+ *     Builds a list of object names from an S3 bucket into a vector. Returns 0 if success,
+ * otherwise 1.
  */
-bool
-S3Connection::ListObjects(const std::string &bucketName, std::vector<std::string> &objects) const
+int
+S3Connection::ListObjects(const std::string &bucketName, const std::string &prefix,
```

Comment:
I don't believe we have a use case for setting the total objects returned other than satisfying the fs interface to return a single entry, correct? We could consider using a boolean `listSingle` input parameter rather than `maxObjects` to simplify the logic and improve readability. Then the case where `listSingle` is true takes only a few lines of code, and otherwise we only need to check the `continuationToken` in the `while` loop. Thoughts??

Reply:
Thanks Tammy! I have changed `ListObjects` to take a `listSingle` boolean input, rather than `maxObjects`, and also updated the tests accordingly.